src/main/java/org/wildfly/security/http/impl/FormAuthenticationMechanism.java

```diff
@@ -90,30 +92,9 @@ public String getMechanismName() {
      */
     @Override
     public void evaluateRequest(final HttpServerRequest request) throws HttpAuthenticationException {
-
-        // Do we have a cached identity?
-        HttpScope session = request.getScope(Scope.SESSION);
-        if (session != null) {
-            FormIdentityCredentials identityCredentials = session.getAttachment(IDENTITY_CREDENTIAL_KEY, FormIdentityCredentials.class);
-            if (identityCredentials != null) {
-                String username = identityCredentials.getUsername();
-                char[] password = identityCredentials.getPassword();
-                try {
-                    if (authenticate(null, username, password) && authorize(username)) {
-                        succeed();
-                        request.authenticationComplete();
-                        request.resumeRequest();
-                        return;
-                    } else {
-                        identityCredentials.dispose();
-                        session.setAttachment(IDENTITY_CREDENTIAL_KEY, null);
-                    }
-                } catch (IOException | UnsupportedCallbackException e) {
-                    throw new HttpAuthenticationException(e);
-                } finally {
-                    fill(password, (char) 0x00);
-                }
-            }
+        // try to re-authenticate a cached principal, if any
+        if (attemptReAuthentication(request)) {
```

Comment:
This PR is just an update based on ELY-585 changes. So we can remove that code to store credentials into session and just store the principal name. The behavior is pretty much the same as before, except that now it is storing the principal name instead of credentials.

Cache is what I'm working now :)